examples/fabric-admin/commands/interactive/InteractiveCommands.cpp

```diff
@@ -31,16 +34,51 @@ constexpr char kInteractiveModeStopCommand[]     = "quit()";
 
 namespace {
 
+// File pointer for the log file
+FILE * logFile = nullptr;
+
+void OpenLogFile(const char * filePath)
+{
+    logFile = fopen(filePath, "a");
+    if (logFile == nullptr)
+    {
+        perror("Failed to open log file");
+    }
+}
+
+void CloseLogFile()
+{
+    if (logFile != nullptr)
+    {
+        fclose(logFile);
+        logFile = nullptr;
+    }
+}
+
 void ClearLine()
 {
     printf("\r\x1B[0J"); // Move cursor to the beginning of the line and clear from cursor to end of the screen
 }
 
 void ENFORCE_FORMAT(3, 0) LoggingCallback(const char * module, uint8_t category, const char * msg, va_list args)
 {
-    ClearLine();
-    chip::Logging::Platform::LogV(module, category, msg, args);
-    ClearLine();
+    uint64_t timeMs       = chip::System::SystemClock().GetMonotonicMilliseconds64().count();
+    uint64_t seconds      = timeMs / 1000;
+    uint64_t milliseconds = timeMs % 1000;
+
+    flockfile(logFile);
+
+    // Portable thread and process identifiers
+    auto pid = static_cast<unsigned long>(getpid());
+    auto tid = static_cast<unsigned long>(pthread_self());
```

Comment:
Simply remove PID and thread id. They are not portable and make assumptions about the platform. They are not important to log.